### ISSUE TITLE
Client Error Bad Request

### ISSUE THREAD

**Issue by rcoletti116** [amickael/eero-adguard-sync#2]
When running the ``eag-sync sync`` I get a client error. The first time I did a sync it worked and synchronized a single client, but now it does not work. Any ideas?

```shell
Traceback (most recent call last):
  File "/usr/local/bin/eag-sync", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/eero_adguard_sync/commands/sync.py", line 260, in sync
    handler.sync(delete, overwrite)
  File "/usr/local/lib/python3.9/dist-packages/eero_adguard_sync/commands/sync.py", line 142, in sync
    self.create(dhcp_diff)
  File "/usr/local/lib/python3.9/dist-packages/eero_adguard_sync/commands/sync.py", line 62, in create
    self.adguard_client.add_client_device(
  File "/usr/local/lib/python3.9/dist-packages/eero_adguard_sync/client/adguard.py", line 67, in add_client_device
    return self.__perform_client_action("control/clients/add", payload)
  File "/usr/local/lib/python3.9/dist-packages/eero_adguard_sync/client/adguard.py", line 61, in __perform_client_action
    resp.raise_for_status()
  File "/usr/local/lib/python3.9/dist-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://10.0.0.76/control/clients/add
```

**Comment by rcoletti116** [comment]
Does the tool support IPv6?  I noticed Adguard is reporting most of my clients by IPv6, and the now 3 successful clients I have sync'd are IPv4 only.

**Comment by rcoletti116** [comment]
@amickael Tagging you in case you haven't seen. Not trying to bother but if you have any insight I would appreciate it. Thank you!

**Comment by amickael** [comment]
@rcoletti116 Thanks for flagging, sorry, I've been traveling the past couple of weeks. I'll take a look this weekend.

**Comment by jaydhary14** [comment]
File "/usr/local/lib/python3.9/dist-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://xxx.xxx.x.xxx/control/clients/add

**Comment by PleaseStopAsking** [comment]
Seeing same error at this time. I am also seeing some deltas between the number of hosts I see within the app vs. the clients reported from eag-sync output. The tool is seeing 27 clients, I have 20 clients at most but when I ran the tool, only 15 where actively connected to the network.

**First Run**
```
Starting sync...
Add new clients  [######################--------------]  17/27
Traceback (most recent call last):
  File "/usr/local/bin/eag-sync", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/commands/sync.py", line 260, in sync
    handler.sync(delete, overwrite)
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/commands/sync.py", line 142, in sync
    self.create(dhcp_diff)
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/commands/sync.py", line 62, in create
    self.adguard_client.add_client_device(
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/client/adguard.py", line 67, in add_client_device
    return self.__perform_client_action("control/clients/add", payload)
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/client/adguard.py", line 61, in __perform_client_action
    resp.raise_for_status()
  File "/usr/local/lib/python3.10/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://<redacted>/control/clients/add
```

**Second Run**

```
Starting sync...
Update existing clients  [####################################]  17/17
Add new clients  [------------------------------------]  0/10
Traceback (most recent call last):
  File "/usr/local/bin/eag-sync", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/commands/sync.py", line 260, in sync
    handler.sync(delete, overwrite)
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/commands/sync.py", line 142, in sync
    self.create(dhcp_diff)
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/commands/sync.py", line 62, in create
    self.adguard_client.add_client_device(
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/client/adguard.py", line 67, in add_client_device
    return self.__perform_client_action("control/clients/add", payload)
  File "/usr/local/lib/python3.10/site-packages/eero_adguard_sync/client/adguard.py", line 61, in __perform_client_action
    resp.raise_for_status()
  File "/usr/local/lib/python3.10/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://<redacted>/control/clients/add
```

**Comment by dennissugar** [comment]
I added some debugging statements to `adguard.py` and it seems to be caused by an `Unnamed device` in the Eero app.

Request:
`{'ids': ['<redacted>'], 'name': None, 'tags': ['device_other'], 'use_global_settings': True, 'use_global_blocked_services': True}`

Response:
`invalid name`

To fix this you could probably just skip over these.

**Comment by nota3k** [comment]
@amickael can bump the version with this fix? cheers https://github.com/mbmccormick/eero-adguard-sync/commit/e3342c902f9cc9623d022ab7c160a92e84faf8ed

**Comment by maquino1985** [comment]
> @amickael can bump the version with this fix? cheers [mbmccormick@e3342c9](https://github.com/mbmccormick/eero-adguard-sync/commit/e3342c902f9cc9623d022ab7c160a92e84faf8ed)

maintainer seems mia but many devices have no nickname but they have a hostname, how hard would it be use hostname instead of nickname from the list of client devices if nickname is empty instead of skipping it entirely?

**Comment by amickael** [comment]
I've pulled in the change. Sorry for the abandonment, I don't have much time to dedicate to these projects as I'd like. I'm happy to add some willing contributors to the project, otherwise I will have to archive.

I also no longer own Eeros so I will not be able to test anything.
